src/main/java/com/codeborne/selenide/collections/ExactTextsCaseSensitiveInAnyOrder.java

```diff
@@ -0,0 +1,43 @@
+package com.codeborne.selenide.collections;
+
+import com.codeborne.selenide.impl.Html;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+public class ExactTextsCaseSensitiveInAnyOrder extends ExactTexts {
+
+  public ExactTextsCaseSensitiveInAnyOrder(String... exactTexts) {
+    super(exactTexts);
+  }
+
+  public ExactTextsCaseSensitiveInAnyOrder(List<String> exactTexts) {
+    super(exactTexts);
+  }
+
+  @Override
+  public boolean test(List<WebElement> elements) {
+    if (elements.size() != expectedTexts.size()) {
+      return false;
+    }
+
+    for (String expectedText : expectedTexts) {
+      boolean found = false;
+      for (WebElement element : elements) {
+        if (Html.text.equalsCaseSensitive(element.getText(), expectedText)) {
```

Comment:
This double-iteration may be ineffective. Every call to `WebDriver.getText()` is slow. 

For better performance, we should first collect all texts to a `List<String>`, and then iterate this list.

Reply:
1. Updated. Got texts from all elements first and then iterated over a list of Strings. 
Since `TextsInAnyOrder` condition is using the same approach, applied the same change there as well.

2. Added several integration tests.